app/controllers/comments_controller.rb

```diff
@@ -53,51 +35,54 @@ class CommentsController < ApplicationController
   ].freeze
 
   @index_subaction_dispatch_table = {
-    target: :show_comments_for_target,
-    pattern: :comment_search,
-    by_user: :show_comments_by_user,
-    for_user: :show_comments_for_user,
-    by: :index_comment
+    by: :index_query_results
   }.freeze
 
-  # Disable cop because method definition prevents a
-  # Rails/LexicallyScopedActionFilter offense
-  # https://docs.rubocop.org/rubocop-rails/cops_rails.html#railslexicallyscopedactionfilter
-  def index # rubocop:disable Lint/UselessMethodDefinition
-    super
-  end
+  ###########################################################
 
   private
 
   def default_index_subaction
-    list_comments
+    list_all
+  end
+
+  # Show list of latest comments. (Linked from left panel.)
+  def list_all
+    query = create_query(:Comment, :all, by: default_sort_order)
+    show_selected_comments(query)
+  end
+
+  def default_sort_order
+    ::Query::CommentBase.default_order
   end
 
   # Show selected list of comments, based on current Query.  (Linked from
   # show_comment, next to "prev" and "next"... or will be.)
-  def index_comment
-    query = find_or_create_query(:Comment, by: params[:by])
+  def index_query_results
+    sorted_by = params[:by].present? ? params[:by].to_s : default_sort_order
+    query = find_or_create_query(:Comment, by: sorted_by)
     show_selected_comments(query, id: params[:id].to_s, always_index: true)
   end
 
-  # Show list of latest comments. (Linked from left panel.)
-  def list_comments
-    query = create_query(:Comment, :all, by: :created_at)
-    show_selected_comments(query)
-  end
-
   # Shows comments by a given user, most recent first. (Linked from show_user.)
-  def show_comments_by_user
-    user = find_or_goto_index(User, params[:by_user].to_s) || @user
+  def by_user
+    user = find_obj_or_goto_index(
+      model: User, obj_id: params[:by_user].to_s,
+      index_path: comments_path
+    )
```

Comment:
This is a change in behavior:
List comments, instead of the current user's comments, If params[:by_user] is not a real user.